app/jobs/whitehall_document_import_job.rb

```diff
@@ -15,10 +15,19 @@ class WhitehallDocumentImportJob < ApplicationJob
   def perform(document_import)
     document_import = WhitehallImporter::Import.call(document_import)
     document_import = WhitehallImporter::Sync.call(document_import)
+
+    if document_import.completed?
+      GdsApi.whitehall_export.document_migrated(document_import.whitehall_document_id)
+    end
+
     document_import.whitehall_migration&.check_migration_finished
   end
 
   def self.handle_error(document_import, error)
+    if document_import.pending?
+      GdsApi.whitehall_export.unlock_document(document_import.whitehall_document_id)
```

Comment:
We probably have to consider what happens if this fails as if we hit an exception here none of our other exception handling code will run.

Reply:
Good catch - I've written an approach to this here https://github.com/alphagov/content-publisher/pull/1746/commits/18a26d567f0b4dfc0fa2b5430e300edc3c195c55